### PR TITLE
Update user info for device and trust changes

### PR DESCRIPTION
**Pull request by jryans**
This ensure the user info panel updates automatically for device and trust
changes.

Fixes https://github.com/vector-im/riot-web/issues/12134`

### CODE REVIEW

File: src/components/views/right_panel/EncryptionPanel.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const EncryptionPanel = ({verificationRequest, member, onClose}) => {
         setRequest(verificationRequest);
     }, [verificationRequest]);
 
-    const [phase, setPhase] = useState(request.phase);
+    const [phase, setPhase] = useState(request && request.phase);
```
Comment (t3chguy) [inline src/components/views/right_panel/EncryptionPanel.js]: the check below is for undefined so this should be `request ? request.phase : undefined`

Comment (jryans) [reply src/components/views/right_panel/EncryptionPanel.js]: I believe that's the same as my version...?

```
-> bob && bob.phase
<- undefined
-> var bob = { phase: "foo" }
<- undefined
-> bob && bob.phase
<- "foo"
```

Comment (t3chguy) [reply src/components/views/right_panel/EncryptionPanel.js]: depends on the value of bob, I guess in this case it works but is implicit:
image

Comment (jryans) [reply src/components/views/right_panel/EncryptionPanel.js]: Fair enough, a lesson for next time. 😄

**Pull request merged**